src/playground/App.js

```diff
@@ -6,41 +6,42 @@ import CrashAlert from "./components/CrashAlert";
 import Footer from "./components/Footer";
 import CodeEditor from "./components/CodeEditor";
 import { Linter, SourceCodeFixer } from "./node_modules/eslint/lib/linter/";
-import { Legacy } from "@eslint/eslintrc/universal";
 import Unicode from "./utils/unicode";
 import Configuration from "./components/Configuration";
 import Split from "react-split";
 import debounce from "./utils/debounce";
 import "./scss/split-pane.scss";
 
-const linter = new Linter();
-const rules = linter.getRules();
+const linter = new Linter({ configType: "flat" });
+const legacyLinter = new Linter({ configType: "eslintrc" });
+const rules = legacyLinter.getRules();
 const ruleNames = Array.from(rules.keys());
 const rulesMeta = Array.from(rules.entries()).reduce((result, [key, value]) => {
     result[key] = value.meta;
     return result;
 }, {});
 
-const defaultParserOptions = {
-    ecmaFeatures: {},
-    ecmaVersion: "latest",
-    sourceType: "script"
+const defaultLanguageOptions = {
+    parserOptions: {
+        ecmaFeatures: {}
+    }
 };
 
 const fillOptionsDefaults = options =>
     (options
         ? {
-            env: {},
             rules: {},
             ...options,
-            parserOptions: {
-                ...defaultParserOptions,
-                ...options.parserOptions
+            languageOptions: {
+                ...options.languageOptions,
+                parserOptions: {
+                    ecmaFeatures: {},
+                    ...options.languageOptions?.parserOptions
+                }
             }
         }
         : {
-            env: { es6: true },
-            parserOptions: defaultParserOptions,
+            languageOptions: defaultLanguageOptions,
             rules: [...rules.entries()].reduce((result, [ruleId, rule]) => {
                 if (rule.meta.docs.recommended) {
                     result[ruleId] = ["error"];
```

Comment:
When I open https://deploy-preview-509--new-eslint.netlify.app/play/ with a clean cache, recommended rules are not enabled, so it looks like this branch never happens. Maybe because `convertLegacyOptionsToFlatConfig` always returns an object?

Reply:
You're right. Pushed a fix.